crates/block-producer/src/block_producer.rs

```diff
@@ -244,83 +249,114 @@ impl BlockProducer {
             }
         }
 
-        // get txs & withdrawal requests from mem pool
-        let (opt_finalized_custodians, block_param) = {
-            let mem_pool = self.mem_pool.lock().await;
-            mem_pool.output_mem_block()?
-        };
-        let deposit_cells = block_param.deposits.clone();
+        let mut mem_block_output_param = OutputParam::default();
+        loop {
```

Comment:
we should define a max retries count, then raise error after meet the condition.

Reply:
Add max retry count.
Now we max try count check and withdrawals/txs zero check.